### PR TITLE
Set revisions on SSO users

### PR DESCRIPTION
**Pull request by rosstimothy**
Updates GitHub users to always have the correct revision set when updating the user. OIDC and SAML users were already updated and pulled in via the e ref update in #32932.

### CODE REVIEW

**Review by Tener — COMMENTED** [review]
Please add context to the change being made. What are revisions, and why do we need to set them?

File: lib/auth/github.go [hunk 1]
```diff
@@ -826,6 +826,7 @@ func (a *Server) createGithubUser(ctx context.Context, p *CreateUserParams, dryR
 				existingUser.GetName())
 		}
 
+		user.SetRevision(existingUser.GetRevision())
```
[the remark this answers is not in the document]
Comment (Tener) [reply lib/auth/github.go:829]: > I'm not sure that we care what revision is returned from `dryRun`.
I think we do care, if only for consistency.

**Comment by Tener** [comment]
> > Please add context to the change being made. What are revisions, and why do we need to set them?
> 
> Revisions are the fundamental aspect of [Optimistic Locking](https://github.com/gravitational/teleport/blob/master/rfd/0126-backend-migrations.md#optimistic-locking).

Thanks for the link. This clarifies some questions, but not all. Quoting the RFD:

> The backend interface will be extended to support new conditional delete and update methods which enforce optimistic locking. Most if not all user facing and editable resources should use the new optimistic locking primitives to prevent losing changes made by a user. Resources which are updated based on presence are likely not a good candidate for conditional operations due to the amount of stress that may put on backends.

I don't see us using the new conditional update method, just the old one with a different value. Reading the RFD I would expect that old methods should continue to work as before. Is this not the case? What exactly happens with an empty Resource field?

File: lib/auth/github.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tener) [reply lib/auth/github.go:829]: Is there anything wrong with improving this code, though?

**Review by Tener — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

**Pull request merged**